src/components/model/Models.svelte

```diff
@@ -0,0 +1,373 @@
+    hasCreatePlanPermission = featurePermissions.plan.canCreate(user);
+    hasDeleteModelPermission = featurePermissions.model.canDelete(user);
+    hasUpdateModelPermission = featurePermissions.model.canUpdate(user);
+    columnDefs = [
+      ...baseColumnDefs,
+      {
+        cellClass: 'action-cell-container',
+        cellRenderer: (params: ModelCellRendererParams) => {
+          const actionsDiv = document.createElement('div');
+          actionsDiv.className = 'actions-cell';
+          new DataGridActions({
+            props: {
+              deleteCallback: params.deleteModel,
+              deleteTooltip: {
+                content: 'Delete Model',
+                placement: 'bottom',
+              },
+              editCallback: params.editModel,
+              editTooltip: {
+                content: 'Edit Model',
+                placement: 'bottom',
+              },
+              hasDeletePermission: hasDeleteModelPermission,
+              hasEditPermission: hasUpdateModelPermission,
+              rowData: params.data,
+            },
+            target: actionsDiv,
+          });
+
+          return actionsDiv;
+        },
+        cellRendererParams: {
+          deleteModel,
+          editModel,
+        } as CellRendererParams,
+        field: 'actions',
+        headerName: '',
+        resizable: false,
+        sortable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        width: 55,
+      },
+    ];
+  }
+
+  onDestroy(() => {
+    resetModelStores();
+  });
+
+  function createPlanWithModel() {
+    if (selectedModel) {
+      goto(`${base}/plans?modelId=${selectedModel.id}`);
+    }
+  }
+
+  function editModel() {
+    if (selectedModel) {
+      goto(`${base}/models/${selectedModel.id}`);
+    }
+  }
+
+  function deleteModel(model: ModelSlim) {
+    effects.deleteModel(model, user);
+  }
+
+  function deleteModelContext(event: CustomEvent<RowId[]>) {
+    const selectedModelId = event.detail[0] as number;
+    const modelToDelete = $models.find((model: ModelSlim) => model.id === selectedModelId);
+    if (modelToDelete) {
+      deleteModel(modelToDelete);
+    }
+  }
+
+  function deselectModel() {
+    selectModel(null);
+  }
+
+  function selectModel(model: ModelSlim | null) {
+    selectedModel = model;
+  }
+
+  async function submitForm(e: SubmitEvent) {
+    if (files) {
+      const newModelId = await effects.createModel(name, version, files, user, description);
+      if ($createModelError === null && e.target instanceof HTMLFormElement) {
+        goto(`${base}/models/${newModelId}`);
+      }
+    }
+  }
+</script>
+
+<PageTitle title="Models" />
+
+<CssGrid columns="20% auto">
+  <Panel borderRight padBody={false}>
+    <svelte:fragment slot="header">
+      <SectionTitle>{selectedModel ? selectedModel.name : 'New Model'}</SectionTitle>
+      {#if selectedModel}
+        <button class="st-button icon fs-6" on:click={deselectModel}>
```

Comment:
Consider adding a tooltip here to indicate that the user is closing vs deleting? I found myself initially struggling to figure out how to get off of the selected model and go back to the upload form. Could also consider an escape key listener to close this as well?

Reply:
Probably going to need @lklyne's input on this for feedback. Sounds like the close button isn't enough? I'm a little hesitant to add an escape key listener because it might need to be a listener on the entire page? I just want to make sure that it doesn't conflict with other interactions.